temoa/types/__init__.py

```
@@ -0,0 +1,318 @@
+RegionPeriodSeasonTimeInputTechVintageOutput = Tuple[
+    Region, Period, Season, TimeOfDay, Commodity, Technology, Vintage, Commodity
+]
+
+# Type aliases for common data structures
+SparseIndex = Union[
+    RegionPeriod,
+    RegionPeriodTech,
+    RegionPeriodTechVintage,
+    RegionPeriodSeasonTimeInputTechVintageOutput,
+    Tuple[Any, ...],
+]
```

Ruff (0.14.0)

Use `X | Y` for type annotations

(UP007)